src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  color: 'red' | 'green' | 'blue';
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
 
-export class Circle {
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    sideA: number,
+    sideB: number,
+    sideC: number,
+  ) {
+    if (
+      sideA <= 0 || sideB <= 0 || sideC <= 0
+    ) {
+      throw new Error('All sides should be positive numbers!');
+    }
 
+    if (
+      sideA + sideB <= sideC
+      || sideB + sideC <= sideA
+      || sideC + sideA <= sideB
+    ) {
+      throw new Error('Those values can\'t form a triangle.');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.sideA = sideA;
+    this.sideB = sideB;
+    this.sideC = sideC;
+  }
+
+  getArea(): number {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+    const area = Math.sqrt(s * (s - this.sideA)
+      * (s - this.sideB) * (s - this.sideC));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: 'circle';
+
+  color: 'red' | 'green' | 'blue';
+
+  radius: number;
 
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    radius: number,
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again